include/matrix_expression_iterator.h

```diff
@@ -0,0 +1,156 @@
+#pragma once
+
+#include <iterator>
+#include "access_category.h"
+
+namespace AMatrix {
+
+template <typename TDataType, typename TExpressionType, std::size_t TCategory>
+class ExpressionRandomAccessIterator
```

Comment:
This default template is essentially dead code in the current implementation, since it is never reached.